ada_feeding_actionselect/ada_feeding_actionselect/policy_service.py

```diff
@@ -0,0 +1,275 @@
+
+        posthoc_cls = import_from_string(posthoc_cls_name)
+        assert issubclass(
+            posthoc_cls, PosthocAdapter
+        ), f"{posthoc_cls_name} must subclass PosthocAdapter"
+
+        posthoc_kwargs = self.get_kwargs("posthoc_kws", "posthoc_kwargs")
+        self.posthoc_adapter = context_cls(**posthoc_kwargs)
+
+        # Initialize and validate the policy
+        self.policy = policy_cls(
+            self.context_adapter.dim, self.posthoc_adapter.dim, **kwargs
+        )
+
+        # Start image subscribers
+        self.image = None
+        self.depth = None
+        self.ros_objs = []
+
+        if self.context_adapter.need_rgb:
+            self.ros_objs.append(self.create_subscription(
+                get_img_msg_type("~/image", self), "~/image", self.rgb_callback, 1
+            ))
+        if self.context_adapter.need_depth:
+            self.ros_objs.append(self.create_subscription(
+                get_img_msg_type("~/depth", self), "~/depth", self.depth_callback, 1
+            ))
+
+        self.ros_objs.append(self.create_service(
+            AcquisitionSelect, "~/action_select", self.select_callback
+        ))
+        self.ros_objs.append(self.create_service(
+            AcquisitionReport, "~/action_report", self.report_callback
+        ))
+
+        self.get_logger().info(f"Policy '{policy_name}' initialized!")
+
+    # Subscriptions
+    def rgb_callback(self, msg: Union[CompressedImage, Image]):
+        """
+        Store rgb image
+        """
+        self.image = ros_msg_to_cv2_image(msg)
+
+    def depth_callback(self, msg: Union[CompressedImage, Image]):
+        """
+        Store depth image
+        """
+        self.depth = ros_msg_to_cv2_image(msg)
+
+    # Services
+    def select_callback(self, request, response):
+        """
+        Implement AcquisitionSelect.srv
+        """
+
+        # Run the context adapter
+        context = self.context_adapter.get_context(
+            request.food_context, self.image, self.depth
+        )
+        if context.size != self.context_adapter.dim:
+            response.status = "Bad Context"
+            return response
+
+        # Run the policy
+        response.status = "Success"
+        response = self.policy.choice(context, response)
+        return response
+
+    def report_callback(self, request, response):
+        """
+        Implement AcquisitionReport.srv
+        """
+
+        # Run the context adapter
+        posthoc = self.posthoc_adapter.get_posthoc(np.array(request.posthoc))
+        if posthoc.size != self.posthoc_adapter.dim:
+            response.status = "Bad Posthoc"
+            response.success = False
+            return response
+
+        # Run the policy
+        response.status = "Success"
+        response.success = True
+        response = self.policy.update(posthoc, request, response)
+        return response
+
+    def get_kwargs(self, kws_root: str, kwarg_root: str) -> Dict:
```

Comment:
Not crucial in this PR, but we should consider making this a generic helper in `ada_feeding.helpers` so we can also use it in `create_action_servers.py`.

Maybe add a TODO for it in-code?